File: BaseLib/src/main/java/com/hjc/baselib/utils/permission/RuntimeRationale.java
package com.hjc.baselib.utils.permission;

import android.app.AlertDialog;
import android.content.Context;
import android.content.DialogInterface;
import android.text.TextUtils;

import com.hjc.baselib.R;
import com.yanzhenjie.permission.Rationale;
import com.yanzhenjie.permission.RequestExecutor;
import com.yanzhenjie.permission.runtime.Permission;

import java.util.List;

/**
 * @Author: HJC
 * @Date: 2019/2/21 11:23
 * @Description: 当权限第一次申请被拒绝后(并未勾选禁止后不再询问), 第二次给出申请理由
 */
public final class RuntimeRationale implements Rationale<List<String>> {

    @Override
    public void showRationale(Context context, List<String> permissions, final RequestExecutor executor) {
        List<String> permissionNames = Permission.transformText(context, permissions);
        String message = context.getString(R.string.message_permission_rationale, TextUtils.join("\n", permissionNames));

        new AlertDialog.Builder(context)
                .setCancelable(false)
                .setTitle(R.string.tip)
                .setMessage(message)
                .setPositiveButton(R.string.resume, new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        executor.execute();
                    }
                })
                .setNegativeButton(R.string.cancel, new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        executor.cancel();
                    }
                })
                .show();
    }
}

File: BaseLib/src/main/java/com/hjc/baselib/utils/permission/RuntimeRationale.kt
package com.hjc.baselib.utils.permission

import android.app.AlertDialog
import android.content.Context
import android.text.TextUtils
import com.hjc.baselib.R
import com.yanzhenjie.permission.Rationale
import com.yanzhenjie.permission.RequestExecutor
import com.yanzhenjie.permission.runtime.Permission

class RuntimeRationale : Rationale<List<String?>?> {

    override fun showRationale(
        context: Context,
        permissions: List<String?>?,
        executor: RequestExecutor
    ) {
        val permissionNames = Permission.transformText(context, permissions)
        val message = context.getString(
            R.string.message_permission_rationale,
            TextUtils.join("\n", permissionNames)
        )
        AlertDialog.Builder(context)
            .setCancelable(false)
            .setTitle(R.string.tip)
            .setMessage(message)
            .setPositiveButton(R.string.resume) { dialog, which -> executor.execute() }
            .setNegativeButton(R.string.cancel) { dialog, which -> executor.cancel() }
            .show()
    }
}